client/my-sites/earn/main.tsx

```diff
@@ -14,28 +14,30 @@ import WordAdsPayments from 'calypso/my-sites/earn/ads/payments';
 import WordAdsEarnings from 'calypso/my-sites/stats/wordads/earnings';
 import WordAdsHighlightsSection from 'calypso/my-sites/stats/wordads/highlights-section';
 import { useSelector } from 'calypso/state';
-import { canAccessWordAds } from 'calypso/state/sites/selectors';
+import { canAccessWordAds, isJetpackSite } from 'calypso/state/sites/selectors';
 import { getSelectedSite } from 'calypso/state/ui/selectors';
 import AdsWrapper from './ads/wrapper';
 import Home from './home';
 import MembershipsSection from './memberships';
+import { Query } from './memberships/index';
 import MembershipsProductsSection from './memberships/products';
 import ReferAFriendSection from './refer-a-friend';
 
 type EarningsMainProps = {
 	section: string;
-	adsProgramName: string;
-	query: string;
+	query: Query;
 	path: string;
 };
 
-const EarningsMain = ( { section, adsProgramName, query, path }: EarningsMainProps ) => {
+const EarningsMain = ( { section, query, path }: EarningsMainProps ) => {
 	const translate = useTranslate();
 	const site = useSelector( ( state ) => getSelectedSite( state ) );
 	const canAccessAds = useSelector( ( state ) => canAccessWordAds( state, site?.ID ) );
+	const isJetpack = useSelector( ( state ) => isJetpackSite( state, site?.ID ) );
+	const adsProgramName = isJetpack ? 'Ads' : 'WordAds';
```

Comment:
Before this PR, adsProgramName was intended to be passed in as a prop. But the only place this component is invoked ([here](https://github.com/Automattic/wp-calypso/blob/ca82988c86b5f288a81b4d9b858f521bc9e7ad20/client/my-sites/earn/controller.ts#L20)), that prop is not passed. And it also wasn't deduced in connect/mapStateToProps. So this value was effectively undefined and not working as expected. 

I've updated the code to fetch this following the same logic as in the Ads wrapper component [here](https://github.com/Automattic/wp-calypso/blob/ca82988c86b5f288a81b4d9b858f521bc9e7ad20/client/my-sites/earn/ads/wrapper.jsx#L365).